### ISSUE TITLE
DefaultParameterFactorySupplier doesn't provide RFC7986 factories

### ISSUE THREAD

**Issue by rfc2822** [ical4j/ical4j#418]
DefaultParameterFactorySupplier instantiates rfc5545 and rfc7986, but only returns the rfc5545 ones. I guess this is a small typo?

Please see https://github.com/ical4j/ical4j/pull/413

**Comment by benfortuna** [comment]
Yup a typo, thanks for the fix. Apologies i didn't look at this sooner.

**Comment by rfc2822** [comment]
No problem, thanks for merging! Will it be in 3.0.x too?

**Comment by benfortuna** [comment]
Yes, the 3.x builds are still built from master. 4.0 alpha builds are from another branch.

Also, I don't currently backport to 2.x unless there is a specific request, although I understand this may still be required for Android? (let me know if that is no longer true).

**Comment by rfc2822** [comment]
> Also, I don't currently backport to 2.x unless there is a specific request, although I understand this may still be required for Android? (let me know if that is no longer true).

For Android <8, it is need theoretically (because Android <8 doesn't come with Java 8 APIs).

DAVx5 however uses the brand-new "Core library desugaring", which brings the Time API (and some others, but nut all) back to old Android versions. So no 2.x needed anymore :)

**Comment by michael-petit** [comment]
Hello, 
Not sure this is the right place to ask my question but this conversation seems to be the most closely related to the problem I face.

I downloaded the latest code form the "develop" branch and added it to my Eclipse environment. After adding all required libraries, I am stuck with an error that is reported by Eclipse in net/fortuna/ical4j/data/DefaultParameterFactorySupplier.java.
 
The error says: 
Type mismatch: cannot convert from List<Content.Factory & ParameterFactory> to List<ParameterFactory<? extends Parameter>>
 
I am completely new to ical4j an restarting java programming after several years... And I must say that I am a bit lost with the use of the Supplier mechanism :-( 

Do you have a direct evident clue of where this error comes from?

**Comment by benfortuna** [comment]
Hi, yes I should have looked into this issue earlier, but I believe it is caused by inconsistent use of generics in the factory implementations. I will apply fixes shortly and let me know to see if the issue is resolved.

Just on the use of `Supplier`, its possibly an edge case, but if someone wanted to use a lambda function to customise the supported components/properties/parameters when parsing, this would be possible:

```java
// only support parsing VEVENT objects..
ContentHandlerContext handlerContext = new ContentHandlerContext().withComponentFactorySupplier(
    () -> { Arrays.asList(new VEvent.Factory()) });
```

**Comment by benfortuna** [comment]
@michael-petit I've committed some fixes for generics to develop branch. Let me know if you still see any build issues.
